src/PointwiseFunctions/AnalyticData/GrMhd/CcsnCollapse.cpp

```diff
@@ -0,0 +1,909 @@
+    const tnsr::I<DataType, 3>& x,
+    tmpl::list<hydro::Tags::Pressure<DataType>> /*meta*/) const {
+  const auto rest_mass_density = get<hydro::Tags::RestMassDensity<DataType>>(
+      variables(vars, x, tmpl::list<hydro::Tags::RestMassDensity<DataType>>{}));
+  return {equation_of_state_.pressure_from_density(rest_mass_density)};
+}
+
+// Specific internal energy
+template <typename DataType>
+tuples::TaggedTuple<hydro::Tags::SpecificInternalEnergy<DataType>>
+CcsnCollapse::variables(
+    const gsl::not_null<IntermediateVariables<DataType>*> vars,
+    const tnsr::I<DataType, 3>& x,
+    tmpl::list<hydro::Tags::SpecificInternalEnergy<DataType>> /*meta*/) const {
+  const auto rest_mass_density = get<hydro::Tags::RestMassDensity<DataType>>(
+      variables(vars, x, tmpl::list<hydro::Tags::RestMassDensity<DataType>>{}));
+
+  // Eventually, below will call for piecewise polytrope or tabulated EOS
+  //  auto temperature = vars->temperature;
+  //  auto electron_fraction = vars->electron_fraction;
+  // return{equation_of_state_.
+  // specific_internal_energy_from_density_and_temperature_and_ye_impl(
+  // rest_mass_density, temperature, electron_fraction)};
+
+  return {equation_of_state_.specific_internal_energy_from_density(
+      rest_mass_density)};
+}
+
+// Electron fraction
+template <typename DataType>
+tuples::TaggedTuple<hydro::Tags::ElectronFraction<DataType>>
+CcsnCollapse::variables(
+    const gsl::not_null<IntermediateVariables<DataType>*> vars,
+    const tnsr::I<DataType, 3>& /*x*/,
+    tmpl::list<hydro::Tags::ElectronFraction<DataType>> /*meta*/) const {
+  return {Scalar<DataType>{vars->electron_fraction.value()}};
+}
+
+// Once temperature is added to hydro tags for the grmhd system,
+// temperature will be stored similar to below
+// // Temperature
+// template <typename DataType>
+// tuples::TaggedTuple<hydro::Tags::Temperature<DataType>>
+// CcsnCollapse::variables(
+//     const gsl::not_null<IntermediateVariables<DataType>*> vars,
+//     const tnsr::I<DataType, 3>& /*x*/,
+//     tmpl::list<hydro::Tags::Temperature<DataType>> /*meta*/) const {
+//   return {Scalar<DataType>{vars->temperature.value()}};
+// }
+
+// Velocity (typical spherical coord system rotating around z axis)
+template <typename DataType>
+tuples::TaggedTuple<hydro::Tags::SpatialVelocity<DataType, 3>>
+CcsnCollapse::variables(
+    const gsl::not_null<IntermediateVariables<DataType>*> vars,
+    const tnsr::I<DataType, 3>& x,
+    tmpl::list<hydro::Tags::SpatialVelocity<DataType, 3>> /*meta*/) const {
+  interpolate_vars_if_necessary(vars);
+  auto spatial_velocity = make_with_value<tnsr::I<DataType, 3>>(x, 0.0);
+  // x and y components will have a radial velocity (from file),
+  // and toroidal velocity (from rotation law) component.
+
+  using std::cos;
+  using std::sin;
+  // Omega = omega0/(1 + (r/A)^2)
+  // vphi = distance_from_rotation_axis * Omega
+  auto rotational_velocity =
```

Comment:
`auto` is probably not what you want. With expression templates `auto` will be the expression class of the RHS, not the resulting value. E.g. if you did `auto c = a + b;` then the type would be `blaze::AddExpression<Vector, Vector>` and then every time you use `c` it will do the math, rather than doing the math only once. The flip side is you also want to minimize (or really eliminate) allocations. So what  we often do is something like `get<2>(spatial_velocity) = /*math for rotational velocity*/;`, then use `get<2>(spatial_velocity)` where you need the rotational velocity for the x, y components, and then overwrite `2` at the end. What components/allocations you can reuse depends on the exact calculation, and sometimes then you end up with say `get<2>(spatial_velocity) *= cool_math;`, but this ultimately makes a difference in heavily multithreaded environments (i.e. everywhere nowadays...) since it eliminates system calls and so a serial bottleneck.